pkg/services/vimmachine.go

```diff
@@ -182,59 +188,54 @@ func (v *VimMachineService) GetHostInfo(c capvcontext.MachineContext) (string, e
 		return vsphereVM.Status.Host, nil
 	}
-	vimMachineCtx.Logger.V(4).Info("VMProvisionedCondition is set to false", "vsphereVM", vsphereVM.Name)
+	log.V(4).Info("VMProvisionedCondition is set to false", "vsphereVM", klog.KRef(vsphereVM.Namespace, vsphereVM.Name))
 	return "", nil
 }
 
-func (v *VimMachineService) findVSphereVM(vimMachineCtx *capvcontext.VIMMachineContext) (*infrav1.VSphereVM, error) {
+func (v *VimMachineService) findVSphereVM(ctx context.Context, vimMachineCtx *capvcontext.VIMMachineContext) (*infrav1.VSphereVM, error) {
 	// Get ready to find the associated VSphereVM resource.
 	vm := &infrav1.VSphereVM{}
 	vmKey := types.NamespacedName{
 		Namespace: vimMachineCtx.VSphereMachine.Namespace,
 		Name:      generateVMObjectName(vimMachineCtx, vimMachineCtx.Machine.Name),
 	}
 	// Attempt to find the associated VSphereVM resource.
-	if err := vimMachineCtx.Client.Get(vimMachineCtx, vmKey, vm); err != nil {
+	if err := v.Client.Get(ctx, vmKey, vm); err != nil {
 		return nil, err
 	}
 	return vm, nil
 }
 
-func (v *VimMachineService) reconcileProviderID(vimMachineCtx *capvcontext.VIMMachineContext, vm *infrav1.VSphereVM) (bool, error) {
+func (v *VimMachineService) reconcileProviderID(ctx context.Context, vimMachineCtx *capvcontext.VIMMachineContext, vm *infrav1.VSphereVM) (bool, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("VSphereVM", klog.KRef(vm.Namespace, vm.Name))
 	biosUUID := vm.Spec.BiosUUID
 	if biosUUID == "" {
-		vimMachineCtx.Logger.Info("spec.biosUUID is empty",
-			"vmKind", vm.Kind,
-			"vmNamespace", vm.Namespace,
-			"vmName", vm.Name)
+		log.Info("spec.biosUUID is empty")
 		return false, nil
 	}
 
 	providerID := infrautilv1.ConvertUUIDToProviderID(biosUUID)
 	if providerID == "" {
-		return false, errors.Errorf("invalid BIOS UUID %s from %s %s/%s for %s",
-			biosUUID,
-			vm.Kind,
-			vm.Namespace,
-			vm.Name,
-			vimMachineCtx)
+		return false, errors.Errorf("invalid BIOS UUID %s for %s", biosUUID, vimMachineCtx)
 	}
 	if vimMachineCtx.VSphereMachine.Spec.ProviderID == nil || *vimMachineCtx.VSphereMachine.Spec.ProviderID != providerID {
 		vimMachineCtx.VSphereMachine.Spec.ProviderID = &providerID
-		vimMachineCtx.Logger.Info("updated provider ID", "provider-id", providerID)
+		log.Info("Updated provider ID", "providerID", providerID)
 	}
 
 	return true, nil
 }
 
 //nolint:nestif
-func (v *VimMachineService) reconcileNetwork(vimMachineCtx *capvcontext.VIMMachineContext, vm *infrav1.VSphereVM) (bool, error) {
+func (v *VimMachineService) reconcileNetwork(ctx context.Context, vimMachineCtx *capvcontext.VIMMachineContext, vm *infrav1.VSphereVM) (bool, error) {
+	log := ctrl.LoggerFrom(ctx)
 	var errs []error
 	networkStatusListOfIfaces := vm.Status.Network
 	var networkStatusList []infrav1.NetworkStatus
 	for i, networkStatusListMemberIface := range networkStatusListOfIfaces {
+		log := log.WithValues("vm", vm.Name, "vmNamespace", vm.Namespace)
```

Comment:
This should not be required when we add it already to the logger and context in the `ReconcileNormal` function :-)
```suggestion
```